java/solomiia_tymoshchuk/src/com/zoolatech/lecture1/tasks/_4/CharCounter.java

```diff
@@ -0,0 +1,36 @@
+package com.zoolatech.lecture1.tasks._4;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class CharCounter {
+    private final String input;
+
+    CharCounter(String input) {
+        this.input = input;
+    }
+
+    public Map<String, Long> countCharsInString() {
+        Map<String, Long> map = Arrays.stream(input.split(""))
```

Comment:
It's better to use `input.toCharArray` instead of `input.split("")` - second method uses regular expressions internally, which can be a cost operation